### PR TITLE
feat(provider/kubernetes): implement lb apis

### PR DESCRIPTION
**Pull request by ethanfrogers**
💀 WIP: DO NOT MERGE 💀 

Implementing the following APIs that were missing:

* `KubernetesLoadBalancerProvider.byAccountAndRegionAndName`
* `KubernetesLoadBalancerProvider.get`

Not required by Deck (yet), but could be useful for external clients
(cli for managing resources, perhaps?)

@lwander @ttomsu PTAL. I could use some guidance on implementing the `get` method. Since all we get is `name`, I'm a little confused on how to differentiate Load Balancers with the same name across accounts/namespaces (regions). I don't know if this is possible in other providers, but is valid in Kubernetes.

Also, should I implement finding the attached server groups? The `getApplicationLoadBalancers` does this, but the `list` method does not. I want to be consistent.

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesLoadBalancerProvider.groovy [hunk 1]
```diff
@@ -121,16 +121,26 @@ class KubernetesLoadBalancerProvider implements LoadBalancerProvider<KubernetesL
     }
   }
 
-  // TODO(lwander): Implement if/when these methods are needed in Deck.
   @Override
   LoadBalancerProvider.Item get(String name) {
-    throw new OperationNotSupportedException("Kubernetes is a special snowflake.")
+    // this is just a placeholder
+    return list().find { it.name == name }
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesLoadBalancerProvider.groovy]: Yeah I'm not sure what this method is supposed to do, any ideas @ttomsu?

Comment (ttomsu) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesLoadBalancerProvider.groovy]: It's a getter based on LB name... where's the confusion?

IIRC, @anotherchrisberry once said that LBs have to have unique names. I think this endpoint might be used to that end.

Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesLoadBalancerProvider.groovy]: Typically you need an account & region to make it unique.

Comment (ethanfrogers) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesLoadBalancerProvider.groovy]: Yep. We have Load Balancers with the same name across multiple clusters so something like `myapp-internal` exists multiple times in the list.

Comment (ethanfrogers) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesLoadBalancerProvider.groovy]: @lwander @ttomsu I _do_ know that `objectMapper.converValue` uses it. I guess it's supposed to get more details attributes?

Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesLoadBalancerProvider.groovy]: Paging the @ttomsu - is it really the case that AWS LBs are unique across all accounts and regions? (even ones you don't own)

Comment (lwander) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesLoadBalancerProvider.groovy]: Or @anotherchrisberry

**Comment by ethanfrogers** [comment]
@lwander FYI, I made progress on this over the weekend. It seems that the `get` method is only called if you don't supply `translateLoadBalancer` with all the data `objectMapper` needs. I have it responding with the Load Balancer but the list of attached server groups isn't working yet.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesLoadBalancerProvider.groovy (hunk 1 above)
Comment (anotherchrisberry) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v1/provider/view/KubernetesLoadBalancerProvider.groovy]: AWS ELB names do not have to be unique across account/regions, but they do need to be unique across all the _VPCs_ in a given account/region, which might be where the confusion is coming from here.

**Comment by ethanfrogers** [comment]
Closing until I have some more time to work on this.

**Pull request closed without merging**